### PR TITLE
Fix transitive resource deps for asset value loader

### PR DESCRIPTION
**Pull request by dpeng817**
Fixes https://github.com/dagster-io/dagster/issues/12854

### CODE REVIEW

File: python_modules/dagster/dagster/_core/storage/asset_value_loader.py [hunk 1]
```diff
@@ -98,7 +99,10 @@ def load_asset_value(
         )
         io_manager_key = assets_def.get_io_manager_key_for_asset_key(asset_key)
         io_manager_def = resource_defs[io_manager_key]
-        required_resource_keys = io_manager_def.required_resource_keys | {io_manager_key}
+        get_transitive_required_resource_keys(io_manager_def.required_resource_keys, resource_defs)
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/storage/asset_value_loader.py:102]: Is this line on purpose?

Comment (dpeng817) [reply python_modules/dagster/dagster/_core/storage/asset_value_loader.py:102]: yes - this is what enables the transitive resource dependencies. They didn't work before this.

Comment (dpeng817) [reply python_modules/dagster/dagster/_core/storage/asset_value_loader.py:102]: nope

**Review by sryza — APPROVED** [review]
nice

**Pull request merged**